src/Modules/Server.cpp

```diff
@@ -138,6 +140,20 @@ DETOUR(Server::PlayerMove)
     auto player = *reinterpret_cast<void**>((uintptr_t)thisptr + Offsets::player);
     auto mv = *reinterpret_cast<const CHLMoveData**>((uintptr_t)thisptr + Offsets::mv);
 
+    if (sar_crosshair_mode.GetBool()) {
+        auto m_hActiveWeapon = entityList->GetEntityIndex(*reinterpret_cast<CBaseHandle*>((uintptr_t)player + Offsets::m_hActiveWeapon));
```

Comment:
LookupEntity should be used like this:
```suggestion
        auto m_hActiveWeapon = *reinterpret_cast<CBaseHandle*>((uintptr_t)player + Offsets::m_hActiveWeapon);
        server->portalGun = entityList->LookupEntity(m_hActiveWeapon);
```